gtfs_funnel/schedule_stats_by_route_direction.py

```diff
@@ -40,28 +128,44 @@ def assemble_scheduled_trip_metrics(
     trip_cols = ["schedule_gtfs_dataset_key", "trip_instance_key"]
     
     grouped_df = df.groupby(trip_cols, observed=True, group_keys=False)
-
+    
+    trips_to_route_cols_subset = ["trip_instance_key", "route_id", "direction_id"]
+    
     # Get median / mean stop meters for the trip
     # Attach time-of-day and route_id and direction_id
-    df2 = pd.merge(
+    # Merge using a subset
+    median_stop_meters_df= pd.merge(
         grouped_df.agg({"stop_meters": "median"}).reset_index().rename(
             columns = {"stop_meters": "median_stop_meters"}),
         time_of_day,
         on = "trip_instance_key",
         how = "left"
     ).merge(
-        trips_to_route,
+        trips_to_route[trips_to_route_cols_subset],
         on = "trip_instance_key",
         how = "inner"
     )
     
-    return df2
+    median_stop_meters_df.direction_id = median_stop_meters_df.direction_id.fillna(0)
+    
+    # Get cardinal direction
+    cardinal_direction_df = find_most_common_dir(df,trips_to_route)
+    
+    # Merge everything together
+    m1 = pd.merge(
+    median_stop_meters_df,
+    cardinal_direction_df,
+    on=["schedule_gtfs_dataset_key", "route_id", "direction_id"],
+    how="inner")
+    
+    return m1
     
     
 def schedule_metrics_by_route_direction(
     df: pd.DataFrame,
     analysis_date: str,
-    group_cols: list
+    group_cols: list,
+    merge_cols:list
```

Comment:
Once your cardinal direction function is added, you no longer need `merge_cols` as an arg.